### PR TITLE
Bugfix: Rename CR kind to KnativeEventing

### PR DESCRIPTION
**Pull request by cardil**
This PR renames eventing CR to be named `KnativeEventing` and be aligned with knative serving.

This is inline with Operation WG decision made on meeting of Dec 10th and should address #18

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,24 +56,23 @@ build your own image.
...
+## The `KnativeEventing` Custom Resource
 
 The installation of Knative Eventing is triggered by the creation of a
-`Eventing` custom resource (CR) as defined by
+`KnativeEventing` custom resource (CR) as defined by
 [this CRD](config/300-eventing-v1alpha1-knativeeventing-crd.yaml). The operator
-will deploy Knative Eventing in the same namespace containing the `Eventing` CR,
-and this CR will trigger the installation, reconfiguration, or removal of the
-knative eventing resources.
+will deploy Knative Eventing in the same namespace containing the
```
Comment (mattmoor) [inline README.md]: Remove trailing whitespace:
```suggestion
will deploy Knative Eventing in the same namespace containing the
```

**Comment by matzew** [comment]
/hold

let's not merge YET

**Comment by matzew** [comment]
We should merge this after we cut for 0.11 - right ? 

Currently we have no such branch:
https://github.com/knative/eventing-operator/tree/v0.11.0

**Comment by cardil** [comment]
> We should merge this after we cut for 0.11 - right ?

Exactly. This should be for `0.12+`

**Comment by n3wscott** [comment]
/lgtm
/approve

When the 0.11 cuts, we can let the hold off

File: config/300-eventing-v1alpha1-knativeeventing-crd.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
-    kind: Eventing
-    listKind: EventingList
-    plural: eventings
-    singular: eventing
+    kind: KnativeEventing
+    listKind: KnativeEventingList
+    plural: knativeeventings
+    singular: knativeeventing
     shortNames:
     - ke
```
Comment (houshengbo) [inline config/300-eventing-v1alpha1-knativeeventing-crd.yaml]: I am now really leaning on to removing this short name. For example, if we have older version of knative eventing operator installed, this short name is occupied by the CRD eventings.operator.knative.dev. After we applying the new CRD knativeeventings.operator.knative.dev also with this same short name, the terrible thing will happen: none of the old or the new CR can be found by the operator controller. If we remove this short name or change it to something else, this issue is gone. @cardil @n3wscott

Comment (cardil) [reply config/300-eventing-v1alpha1-knativeeventing-crd.yaml]: That's a valid concern for me. I'm okay with removing it.

**Comment by matzew** [comment]
@cardil so, we have https://github.com/knative/eventing-operator/tree/v0.11.0  as the release for 0.11 was cut.

wanna move forward here, cancel the HOLD ?

**Comment by cardil** [comment]
@matzew Please cancel the hold. Also, waiting for lgtm after additional remove of short name.

**Pull request merged**